nautobot/extras/models/roles.py

```diff
@@ -0,0 +1,51 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from nautobot.extras.fields import ForeignKeyLimitedByContentTypes
+from nautobot.extras.models.mixins import SetFieldColorAndDisplayMixin
+from nautobot.extras.models.name_color_content_types import NameColorContentTypesModel
+from nautobot.extras.utils import RoleModelsQuery, extras_features
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class Role(NameColorContentTypesModel):
```

Comment:
Same as the abstract base class, is there any possibility we could have a "freeze" for all new models being added to extras, since we won't be able to do #2722 in 2.0?

Reply:
Is it possible for this to live in Core?